### ISSUE TITLE
Using imgix.js through npm throws some errors

### ISSUE THREAD

**Issue by rmngrc** [imgix/imgix.js#109]
Hi folks,

I'm trying to use imgix.js through npm in a website using typescript and browserify. So, after doing `npm i imgix.js --save` I went to the main file of my application and placed:

```
let imgix = require('imgix.js');
```

Ran the typescript compiler and got some errors:

```
Error: Cannot find module './util.js' from '/Users/ramon/Sites/cms/node_modules/imgix.js/dist'
Error: Cannot find module './util.js' from '/Users/ramon/Sites/cms/node_modules/imgix.js/dist'
Error: Cannot find module './ImgixTag.js' from '/Users/ramon/Sites/cms/node_modules/imgix.js/dist'
Error: Cannot find module './util.js' from '/Users/ramon/Sites/cms/node_modules/imgix.js/dist'
Error: Cannot find module './defaultConfig' from '/Users/ramon/Sites/cms/node_modules/imgix.js/dist'
```

Which make sense if you have a look at the dist files:

```
var util = require('./util.js'),
    targetWidths = require('./targetWidths.js');
```

and

```
var e=t("./util.js"),s=t("./targetWidths.js");
```

I think that the dist files should include the dependencies minified in them or at least they should resolve to them properly. For now I can only get it working importing the src files as `require('imgix.js/src/imgix.js')`.

Thank you,

**Comment by paulstraw** [comment]
Hey @rmngrc, imgix.js itself is compiled with Browserify, which is why those `require` statements are in there. All the `require`d files are actually inlined by Browserify, and it passes in its own `require` function. The solution here would be either:

1. Importing the file from `src`, as you mentioned.
2. Telling your Browserify config that imgix.js has already been run through Browserify using the `noParse` configuration option, i.e. `noParse: ['imgix.js']`.

That should do the trick. Please feel free to reopen this issue if you run into more problems, or have more questions about this!

**Comment by pushred** [comment]
Hi, just wanted to report that the `noParse` instructions are incorrect, it requires an absolute path rather than just the name of the module. For me anyway, this:

```js
browserify(file.path, { noParse: ['imgix.js'] })
```

fails with an `Error: Cannot find module './util.js'` exception while:

```js
browserify(file.path, { noParse: [require.resolve('imgix.js')] })
```

.. does not. I verified this with the CLI as well. This is with browserify 13.1.1 (latest).

See also:

 - https://github.com/substack/module-deps#var-d--mdepsopts
 - https://github.com/substack/node-browserify/issues/1642

`imgix.js/src/imgix.js` does work as-is, this seems like the better way to go altogether? Is the noParse suggestion to avoid possible user build issues? First I've encountered that in a lib I've used with Browserify.
